test/end2end_test.go

```diff
@@ -635,13 +658,24 @@ func (te *test) listenAndServe(ts testpb.TestServiceServer, listen func(network,
 	}
 	sopts = append(sopts, te.customServerOptions...)
 	s := grpc.NewServer(sopts...)
-	te.srv = s
-	if te.healthServer != nil {
-		healthgrpc.RegisterHealthServer(s, te.healthServer)
+
+	// Create a new default health server if enableHealthServer is set, or use
+	// the provided one.
+	var hs healthpb.HealthServer
+	if ts != nil {
+		if tss, ok := ts.(*testServer); ok {
+			if tss.enableHealthServer {
+				hs = health.NewServer()
+				healthgrpc.RegisterHealthServer(s, hs)
+			}
+		}
+		testpb.RegisterTestServiceServer(s, ts)
 	}
-	if te.testServer != nil {
-		testpb.RegisterTestServiceServer(s, te.testServer)
+	if te.healthServer != nil {
```

Comment:
This is no longer necessary, right?
If you need the health server, you can read `test.hSrv`.

Reply:
Done.